processing/src/main/java/io/druid/query/topn/TopNLexicographicResultBuilder.java

```diff
@@ -122,9 +183,18 @@ public int compare(
               DimValHolder o2
           )
           {
-            return comparator.compare(o1.getDimName(), o2.getDimName());
+            return comparator.compare(o2.getDimName(), o1.getDimName());
           }
         }
-    ).maximumSize(threshold).create();
+    );
   }
+
+  private boolean shouldAdd(String dimName)
```

Comment:
Is it possible to avoid copy pasting the code and reuse the code from the other topnresultbuilders?